syntax = "proto3";

package core.pb;

message Vec2f
{
    float x = 1;
    float y = 2;
}

message Vec3f
{
    float x = 1;
    float y = 2;
    float z = 3;
}

message Vec4f
{
    float x = 1;
    float y = 2;
    float z = 3;
    float w = 4;
}

message Vec2i
{
    int32 x = 1;
    int32 y = 2;
}

message Vec3i
{
    int32 x = 1;
    int32 y = 2;
    int32 z = 3;
}

message Vec4i
{
    int32 x = 1;
    int32 y = 2;
    int32 z = 3;
    int32 w = 4;
}